src/app/home/home.component.spec.ts

```diff
@@ -1,45 +1,49 @@
 import { TestBed, async, ComponentFixture } from '@angular/core/testing';
-import { DebugElement } from '@angular/core';
+import { DebugElement, Component, Input } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { RouterTestingModule } from '@angular/router/testing';
 import { of } from 'rxjs';
-import { NoopAnimationsModule } from '@angular/platform-browser/animations';
-
-import { SharedModule } from '../shared/shared.module';
 
 import { MockHalService } from 'ngx-prx-styleguide';
-import { CampaignListService } from '../campaign-list/campaign-list.service';
-import { CampaignListModule } from '../campaign-list/campaign-list.module';
+import { CampaignListService, CampaignParams } from '../campaign-list/campaign-list.service';
 import { CampaignListServiceMock, params } from '../campaign-list/campaign-list.service.mock';
 import { HomeComponent } from './home.component';
 
```

Comment:
Thought I had the material imports sorted out and coming from the SharedModule, but this test is where things really fell apart. (It's just such a pain to reimport everything everywhere and keep track of it all.) So what I did instead was go back to importing material close to where it is used and isolate this parent component test it by mocking out the component it contains.